apps/browser-extension-wallet/src/lib/scripts/background/services/cardanoTokenPrices.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { Subject } from 'rxjs';
+import { storage } from 'webextension-polyfill';
+import { TokenPrices, Status } from '../../types';
+import { Wallet } from '@lace/cardano';
+import { Cardano } from '@cardano-sdk/core';
+import { config } from '@src/config';
+
+/** Shortcut for the token prices subject type. */
+type TokenPriceSubject = Subject<{ tokens: TokenPrices } & Status>;
```

Comment:
```suggestion
type TokenPriceSubject = BehaviorSubject<{ tokens: TokenPrices } & Status>;
```

This way you don't need the local priceData variable because you can access the last emited value: `priceSubject$.value`;

Reply:
This is just an interface: the `Observable` used in the implementation is (ATM) actually a `BehaviorSubject`.
In case in the future we need to change the actual implementation, using the ancestor class `Subject` reduces the chances we need to change this interface type as well.

Reply:
Sorry, I was not clear enough. I meant:
```
// have a local BehaviorSubject
const price$ = new BehaviorSubject<{ tokens: TokenPrices } & Status>();
...
const updatePriceData = (assetId: string, data?: PriceData) => {
  const now = Date.now();
  const nextPriceData = {
    ...price$.value,  // access current value from the subject
    [assetId]: data ? [now, data] : [now],
  };
  emitPrices(nextPriceData);
...
const initCardanoTokenPrices = () => {
  storage.local
    .get(CACHE_KEY)
    .then((data) => {
      const initialPriceData = data[CACHE_KEY] || {};
      emitPrices(initialPriceData);
      fetchPrices();
  return price$;  // expose local subject
};
```

Then it could be assigned to coinPrices.tokenSubject$ in utilityService.ts
```
const coinPrices: CoinPrices = {
  tokenPrices$: initCardanoTokenPrices(),
};
```

But again, this is just a suggestion. I am fine with either way.

Reply:
Nice suggestion! Thank you.

5b33fc6db should solve this.